src/garage/torch/_functions.py

```diff
@@ -383,6 +383,61 @@ def state_dict_to(state_dict, device):
     return state_dict
 
 
+# pylint: disable=abstract-method
+class Module(nn.Module):
+    """Wrapper class for Garage PyTorch modules."""
+
+    def __getstate__(self):
+        """Save the current device of the module before saving module state.
+
+        Returns:
+            dict: State dictionary.
+        """
+        # do we always run experiments on global device?
+        save_from_device = global_device()
+        self.to('cpu')
+        state = self.__dict__.copy()
+        state['device'] = save_from_device
+        return state
+
+    def __setstate__(self, state):
+        """Restore the module state, moving it back to the original device if possible.
+
+        Args:
+            state (dict): State dictionary.
+
+        """
+        system_device = global_device()
+        save_from_device = state['device']
+        if save_from_device == system_device:
+            module_state_to(state, system_device)
```

Comment:
I think you can use the other state-dict moving function you wrote here (even though this kinda isn't a state dict).

Reply:
Will every movable parameter of a nn.module be a tensor or a dict?